src/server/middleware/manifestinfo/manifest_info.go

```diff
@@ -0,0 +1,74 @@
+package manifestinfo
+
+import (
+	"fmt"
+	"github.com/goharbor/harbor/src/common/utils"
+	ierror "github.com/goharbor/harbor/src/internal/error"
+	project2 "github.com/goharbor/harbor/src/pkg/project"
+	"github.com/goharbor/harbor/src/server/middleware"
+	reg_err "github.com/goharbor/harbor/src/server/registry/error"
+	"github.com/opencontainers/go-digest"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+var (
+	manifestURLRe = regexp.MustCompile(`^/v2/((?:[a-z0-9]+(?:[._-][a-z0-9]+)*/)+)manifests/([\w][\w.:-]{0,127})`)
+)
+
+// Middleware gets the manifest information from request and inject it into the context
+func Middleware() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			mf, err := parseManifestInfoFromPath(req)
+			if err != nil {
+				reg_err.Handle(rw, req, err)
+				return
+			}
+			*req = *(req.WithContext(middleware.NewManifestInfoContext(req.Context(), mf)))
+			next.ServeHTTP(rw, req)
```

Comment:
`*req = *(req.WithContext(middleware.NewManifestInfoContext(req.Context(), mf)))` not needed just `next.ServeHTTP(rw, req.WithContext(...))` will be fine.